pkg/controllers/management/auth/globalroles/fleetworkspace_role_handler.go

```diff
@@ -0,0 +1,162 @@
+// - 1) CR for ResourceRules
+// - 2) CR for WorkspaceVerbs
+type fleetWorkspaceRoleHandler struct {
+	crClient rbacv1.ClusterRoleController
+	crCache  rbacv1.ClusterRoleCache
+	fwCache  mgmtcontroller.FleetWorkspaceCache
+}
+
+func newFleetWorkspaceRoleHandler(management *config.ManagementContext) *fleetWorkspaceRoleHandler {
+	return &fleetWorkspaceRoleHandler{
+		crClient: management.Wrangler.RBAC.ClusterRole(),
+		crCache:  management.Wrangler.RBAC.ClusterRole().Cache(),
+		fwCache:  management.Wrangler.Mgmt.FleetWorkspace().Cache(),
+	}
+}
+
+// ReconcileFleetWorkspacePermissions reconciles backing ClusterRoles created for granting permission to fleet workspaces.
+func (h *fleetWorkspaceRoleHandler) reconcileFleetWorkspacePermissions(globalRole *v3.GlobalRole) error {
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil &&
+		globalRole.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+		return nil
+	}
+
+	if err := h.reconcileResourceRules(globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet permissions cluster role: %w", err)
+	}
+	if err := h.reconcileWorkspaceVerbs(globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet workspace verbs cluster role: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceRoleHandler) reconcileResourceRules(globalRole *v3.GlobalRole) error {
+	crName := wrangler.SafeConcatName(globalRole.Name, fleetWorkspaceClusterRulesName)
+	cr, err := h.crCache.Get(crName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if apierrors.IsNotFound(err) {
+		_, err := h.crClient.Create(&v1.ClusterRole{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crName,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: v3.GlobalRoleGroupVersionKind.GroupVersion().String(),
+						Kind:       v3.GlobalRoleGroupVersionKind.Kind,
+						Name:       globalRole.Name,
+						UID:        globalRole.UID,
+					},
+				},
+				Labels: map[string]string{
+					grOwnerLabel:                wrangler.SafeConcatName(globalRole.Name),
+					controllers.K8sManagedByKey: controllers.ManagerValue,
+				},
+			},
+			Rules: globalRole.InheritedFleetWorkspacePermissions.ResourceRules,
+		})
+
+		return err
+	}
+
+	if !equality.Semantic.DeepEqual(cr.Rules, globalRole.InheritedFleetWorkspacePermissions.ResourceRules) {
+		// undo modifications if cr has changed
+		cr.Rules = globalRole.InheritedFleetWorkspacePermissions.ResourceRules
+		_, err := h.crClient.Update(cr)
+
+		return err
+	}
+
+	return err
+}
+
+func (h *fleetWorkspaceRoleHandler) reconcileWorkspaceVerbs(globalRole *v3.GlobalRole) error {
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil || len(globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs) == 0 {
+		return nil
+	}
+	crName := wrangler.SafeConcatName(globalRole.Name, fleetWorkspaceVerbsName)
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
```

Comment:
Let's just skip the backing `ClusterRole` creation if there are no fleetworkspaces besides local. Keep in mind this is granting permission to just the cluster-wide fleetworkspace resource, so they wouldn't have access to the `fleet-local` workspace namespace. This is an edge case because the `GlobalRole` would need to exist before the workspaces are created. Maybe that might happen when doing a back up and restore? 
It should be fix in this [commit](https://github.com/rancher/rancher/pull/44901/commits/047060c586a486a6f6205fdadd19dd10567ca7b7), please have a look.